test/library/packages/UnitTest/UnitTest.chpl

```diff
@@ -0,0 +1,677 @@
+module UnitTest {
+  use Reflection;
+  /*Assertion Error class. Raised when assert Function Failed*/
+  class AssertionError: Error {
+    var details: string;
+
+    proc init(details: string = "") {
+      this.details = details;  
+    }
+
+    override proc message() {
+      return this.details;
+    }
+  }
+  /*
+    Assert that a boolean condition is true.  If it is false, prints
+    'assert failed' and rasies AssertionError. 
+
+    :arg test: the boolean condition
+    :type test: `bool`
+  */
+  pragma "insert line file info"
+  pragma "always propagate line file info"
+  proc assertTrue(test: bool) throws {
+    if !test then
+      throw new owned AssertionError("assertTrue failed. Given expression is False");
+  }
+
+  /*
+    Assert that a boolean condition is false.  If it is true, prints
+    'assert failed' and raises AssertionError.
+
+    :arg test: the boolean condition
+    :type test: `bool`
+  */
+  pragma "insert line file info"
+  pragma "always propagate line file info"
+  proc assertFalse(test: bool) throws {
+    if test then
+      throw new owned AssertionError("assertFalse failed. Given expression is True");
+  }
+  
+  /*Function to call the respective method for equality checking based on the type of argument*/
+  private
+  proc checkAssertEquality(first, second) throws {
+    type firstType = first.type;
+    type secondType = second.type;
+
+    if isTupleType(firstType) && isTupleType(secondType) {
+      // both are tuples.
+      assertTupleEqual(first, second);
+      return;
+    }
+    else if isArrayType(firstType) && isArrayType(secondType) {
+      // both are arrays
+      assertArrayEqual(first, second);
+      return;
+    }
+    else if isRangeType(firstType) && isRangeType(secondType) {
+      // both are Range
+      assertRangeEqual(first, second);
+    }
+    else if isString(first) && isString(second) {
+      // both are Strings
+      assertStringEqual(first, second);
+    }
+    else {
+      __baseAssertEqual(first, second);
+    }
+  }
+
+  /*
+    Check that a boolean array is true.  If any element is false, returns 'false'
+    else return 'true'.
+
+    :arg it: the iterator to the array
+    :type it: `iterator`
+  */
+  private
+  proc all(it: _iteratorRecord) {
+    for b in it do if b == false then return false;
+    return true;
+  }
+  /* Method overloading for the above function. Return the argument itself
+  */
+  private
+  proc all(check: bool) {
+    return check;
+  }
+
+  /*An equality assertion for sequences (like arrays, tuples, strings, range).
+    Args:
+      seq1: The first sequence to compare.
+      seq2: The second sequence to compare.
+      seq_type_name: The name of datatype of the sequences
+  */
+  private
+  proc assertSequenceEqual(seq1, seq2, seq_type_name) throws {
+    var tmpString: string;
+    var len1: int = seq1.size;
```

Comment:
Some stylistic tips (all optional) :

- It's OK to rely on the type inference for declarations
- It can be helpful to make variables `const` if you don't expect them to change, for safety and sometimes performance reasons.
- you can do comma-separated declarations, especially useful for comma-separated declarations

So this could look like:

```chpl
const len1 = seq1.size,
      len2 = seq2.size;
```

Reply:
Sure, I think it's best to have these as const. I will make the necessary changes.

Reply:
I have made the changes and will follow similar coding style.